### PR TITLE
Change the locally defined Db's to use aliases from the ipmiComm db's

### PR DESCRIPTION
**Pull request by gustavosr8**
Since the defined Db's are just copies of ipmiComm db's, with the name changed to fit the LNLS name convention, we can replace the record definition with aliases to the support module record.

### CODE REVIEW

File: ipmiMgrApp/Db/cooling_unit.substitutions [hunk 1]
```diff
@@ -23,19 +23,48 @@
...
+            { 12V          , 3           , 2       , $(fruid) , $(id)  }
+            { Fan1         , 1           , 4       , $(fruid) , $(id)  }
+            { Fan2         , 2           , 4       , $(fruid) , $(id)  }
+            { Fan3         , 3           , 4       , $(fruid) , $(id)  }
+            { Fan4         , 4           , 4       , $(fruid) , $(id)  }
+            { Fan5         , 5           , 4       , $(fruid) , $(id)  }
+            { Fan6         , 6           , 4       , $(fruid) , $(id)  }
+}
+
+file "sensor_ai_alias.db"
```
Comment (henriquesimoes) [inline ipmiMgrApp/Db/cooling_unit.substitutions]: I really don't know if this pays off, but is it possible to have a substitution file that substitutes the filename from the `file` clause while keeping anything else untouched?

This could avoid having aliases duplicating all expansions. Surely would increase significantly the amount of files and complexity of the substitutions graph.

File: ipmiMgrApp/Db/cooling_unit.substitutions [hunk 2]
```diff
@@ -49,4 +78,5 @@ file "sensor_ai.db"
             { Fan4         , 4           , 4       , $(fruid) , $(id)  }
             { Fan5         , 5           , 4       , $(fruid) , $(id)  }
             { Fan6         , 6           , 4       , $(fruid) , $(id)  }
-}
\ No newline at end of file
+}
+
```
Comment (henriquesimoes) [inline ipmiMgrApp/Db/cooling_unit.substitutions]: Nit: useless empty.

File: ipmiMgrApp/Db/sensor_ai_alias.template [hunk 3]
```diff
@@ -0,0 +1,3 @@
+alias("$(dev):$(prefix)$(attr)$(sensinst)", "$(P)$(R)$(prefix)$(attr)-Mon")
+
+alias("$(dev):$(prefix)$(attr)$(sensinst)P", "$(P)$(R)$(prefix)$(attr)Prs-Sts")
```
Comment (henriquesimoes) [inline ipmiMgrApp/Db/sensor_ai_alias.template]: Missing `$(sensinst)` on the alias name. As we discussed in person, mention this fixes a bug in the previous implementation where records with the same name would be generated and one would overwrite the other.

File: iocBoot/iocIpmiMgr/stIpmiMgr.cmd [hunk 4]
```diff
@@ -13,9 +13,8 @@ drvAsynIPPortConfigure ("$(PORT)","$(IPADDR):623 udp",0,0,0)
 mchInit("$(PORT)")
 
 ## Load record instances
-dbLoadRecords("$(ASYN)/db/asynRecord.db","P=${P}, R=${R}asyn,PORT=$(PORT),ADDR=0,OMAX=80,IMAX=80")
-dbLoadRecords("${TOP}/db/microtca_bpm_crate.db", "link=$(PORT),P=$(P),R=$(R)")
-
+dbLoadRecords("$(TOP)/db/asynRecord.db","P=${P}, R=${R}asyn,PORT=$(PORT),ADDR=0,OMAX=80,IMAX=80")
+dbLoadRecords("${TOP}/db/microtca_bpm_crate.db", "link=$(PORT),P=$(P),R=$(R),dev=$(P)$(R),unit='',aliasprefix=''")
```
Comment (henriquesimoes) [inline iocBoot/iocIpmiMgr/stIpmiMgr.cmd]: Maybe we can have right-sided `$(P)` (or `$(R)`) defined without the final `:` so that the `dev` does not contain duplicated `::`.

**Review by henriquesimoes — CHANGES_REQUESTED** [review]

File: ipmiMgrApp/Db/cooling_unit.substitutions (hunk 1 above)
Comment (gustavosr8) [reply ipmiMgrApp/Db/cooling_unit.substitutions]: Unfortunately, the substitutions file only generates `.db` files, so it's not possible to expand a `.substitutions` macro in another `.substitutions` file.

File: ipmiMgrApp/Db/sensor_ai_alias.template (hunk 3 above)
Comment (ericonr) [reply ipmiMgrApp/Db/sensor_ai_alias.template]: Honestly, I think most of our IOCs should use [`dbRecordsOnceOnly = 1`](https://github.com/lnls-dig/afc-epics-ioc/blob/master/iocBoot/iocutca/common.cmd#L14).

It only allows records to be defined once, so if your substitutions have any issues, the IOC warns you.

**Review by ericonr — COMMENTED** [review]
I wonder if, in order to avoid duplicating the entries in `substitutions`, you couldn't have the alias files `include` the upstream DBs before defining their aliases.

File: ipmiMgrApp/Db/Makefile [hunk 5]
```diff
@@ -36,6 +34,9 @@ DB += rtm_lamp.db
 #System
 DB += microtca_bpm_crate.db
 
+
+#Modules
+DB_INSTALLS += $(ASYN)/db/asynRecord.db
```
Comment (henriquesimoes) [inline ipmiMgrApp/Db/Makefile]: This change does not belong in this PR, IMHO. b51ac623288a7ac500bec726aeb51cb6483dcd4e already does this in the appropriate context.

File: ipmiMgrApp/Db/afcv3.substitutions [hunk 6]
```diff
@@ -53,4 +54,5 @@ file "sensor_ai.db"
             { TempClkSwitch   , 3           , 1       , $(fruid)     , $(id)  }
             { TempDCDC        , 4           , 1       , $(fruid)     , $(id)  }
             { TempRAM         , 5           , 1       , $(fruid)     , $(id)  }
-}
\ No newline at end of file
+}
+
```
Comment (henriquesimoes) [inline ipmiMgrApp/Db/afcv3.substitutions]: Nit: _remove_ extra empty line. You may want to check this in the other files as well; several of them have extra empty line where shouldn't be needed. Not sure if my previous comment on the topic was ambiguous about adding or removing them. Sorry if so.

File: ipmiMgrApp/Db/afcv3.substitutions [hunk 7]
```diff
@@ -27,13 +27,14 @@
 #  19   Compact  0xf0    0xc1  0x69  0x10              HS 013 AMC9
 # ------------------------------------------------------------------
 
-file "fru_common.db"
+
+file "fru_basic_alias.db"
```
Comment (henriquesimoes) [inline ipmiMgrApp/Db/afcv3.substitutions]: Doesn't `fru_basic_alias.db` already include the expansion of `sensor_ai.db`, through the expansion of the upstream [`fru_basic.template`][1]?

As far as I can understand, the previous `fru_common.db` would not include them. Am I missing something?

[1]: https://github.com/slac-epics-modules/ipmiComm/blob/R4.6.0/Db/fru_basic.substitutions#L32

File: ipmiMgrApp/Db/fru_cu_alias.template [hunk 8]
```diff
@@ -0,0 +1,7 @@
+include "fru_cu.db"
```
Comment (henriquesimoes) [inline ipmiMgrApp/Db/fru_cu_alias.template]: Isn't it [`fru_cu_common.db`][1] in the upstream?

I think, in the end, this is basically the same question I've made regarding `fru_basic_alias`, since `fru_cu.db` includes the substituted records from `sensor_ai.db` while `fru_cu_common.db` does not.

[1]: https://github.com/slac-epics-modules/ipmiComm/blob/R4.6.0/Db/fru_cu_common.db

File: ipmiMgrApp/Db/fru_basic_alias.template [hunk 9]
```diff
@@ -0,0 +1,27 @@
+include "fru_basic.db"
```
Comment (henriquesimoes) [inline ipmiMgrApp/Db/fru_basic_alias.template]: Upstream [`fru_common.db`][1]?

[1]: https://github.com/slac-epics-modules/ipmiComm/blob/R4.6.0/Db/fru_common.db

**Review by henriquesimoes — CHANGES_REQUESTED** [review]
Theses substitutions turn out to be very confusing... I think I found some issues on the current implementation. Maybe I'm simply confusing things. If the latter turns out to be the case, it may payoff to further explain the logic in the commit message.

File: ipmiMgrApp/Db/microtca_bpm_crate.substitutions [hunk 10]
```diff
@@ -12,17 +12,12 @@
...
+    pattern { id, location }
+            { "", "" }
 }
 
-#file "shelf.db"
-#{
-#    pattern { id }
-#            { Shelf }
-#}
-
```
Comment (henriquesimoes) [inline ipmiMgrApp/Db/microtca_bpm_crate.substitutions:25]: This should be in another commit.

File: ipmiMgrApp/Db/shelf.substitutions
Comment (henriquesimoes) [inline ipmiMgrApp/Db/shelf.substitutions:1]: If we don't load this file, why do we care about it?

Remove it along with its commented code in a standalone commit before this one.

**Review by henriquesimoes — CHANGES_REQUESTED** [review]

**Review by henriquesimoes — APPROVED** [review]
Looks good for me. I haven't double-checked aliases names, but they should be fine if you replaced the `record(` by an `alias(` for specifying them all before adding the upstream-used names.

File: ipmiMgrApp/Db/cooling_unit.substitutions [hunk 11]
```diff
@@ -23,19 +23,19 @@
...
 
-file "fru_cu_common.db"
+file "fru_common_alias.db"
 {
     pattern { id    , fruid    }
             { $(id) , $(fruid) }
 }
 
-file "fru_common.db"
+file "fru_cu_common_alias.db"
```
Comment (ericonr) [inline ipmiMgrApp/Db/cooling_unit.substitutions]: This diff is confusing because it's changing cu_common to common and vice-versa. Please reorder this block.

**Review by ericonr — COMMENTED** [review]
Were `system_powerstate_unsupported`, `system_chassis_status`, `fru_extended` templates removed because they were included but not used? Or were they removed as part of the last commit? If the former, I think they should be cleaned up in a separate commit.
